meson.build

```diff
@@ -0,0 +1,121 @@
+## Experimetal meson build system for dinit
+project(
+    'dinit', 
+    'cpp', 
+    version : run_command(
+    [find_program('sh'), 'build/general_variables_for_meson.sh', 'version'],
+    check : true
+    ).stdout(),
+    license : 'Apache',
+    meson_version : '>= 0.47.0',
+    default_options : ['cpp_std = c++11',
+                       'b_lto = true',
```

Comment:
drop, same with the prefix

Reply:
What? i know `prefix` is pointless and must be removed but you marked line 9-12!

Reply:
i marked the lto line, that's user choice, not a default

Reply:
dinit uses a combination of a special configuration file (`mconfig`) and `make` to build itself. `mconfig`s are usually categorized based on the kernel of oses (I think you know that). Each `mconfig` is written to provide the best performance for that target system, in addition to specifying the necessary specifications for that target. For example, all mconfig's `CXXOPTS` contains `-Os` for reduse dinit's execs size by default (if you think thats not good, we can talk about it).
Of course, I may remove `b_lto` in the future, but I will still apply some optimizations by default (and when its necessary, allow users to disable it).

Reply:
the mconfigs exist because there is no kind of configure script

the choice whether to use LTO or not, same with pretty much all other flags that only affect binary size or performance but not functionality, is entirely user choice and should be left to the user, there is no need to clutter the build system with it

Reply:
meson has options you can pass at configure time for this reason and the CXXFLAGS environment variable likewise exists for a reason